src/lib.rs

```diff
@@ -155,10 +166,37 @@ pub fn init_with_rng<T: Rng>(rng: &mut T) -> Result<(), Error> {
 }
 
 /// Produces a 256-bit crypto hash out of the provided `data`.
-pub fn hash(data: &[u8]) -> [u8; 32] {
+pub fn hash(data: &[u8]) -> [u8; HASH_BYTES] {
     hashing_impl::sha3_256(data)
 }
 
+/// Uses Scrypt key derivation to securely derive a byte vector from a `password` and `salt`.
+/// `output` may be used to construct any of the keys in this library.
+///
+/// `output` must satisfy the following condition: `output.len() > 0 && output.len() <= (2^32 - 1) *
+/// 32`.
+///
+/// # Example
+///
+/// ```
+/// use safe_crypto::*;
+///
+/// let password = b"password!";
+/// let salt = b"salt!";
+/// let mut output = [0; SYMMETRIC_KEY_BYTES];
+///
+/// derive_bytes(password, salt, &mut output).unwrap();
+/// let key1 = SymmetricKey::from_bytes(output);
+/// derive_bytes(password, salt, &mut output).unwrap();
+/// let key2 = SymmetricKey::from_bytes(output);
+///
+/// assert_eq!(key1, key2);
+/// ```
+pub fn derive_bytes(password: &[u8], salt: &[u8], output: &mut [u8]) -> Result<(), Error> {
```

Comment:
Can we name this better ? Doesn't seem to convey a whole lot sp. since it's not under any module but directly under lib.rs. Probably call `derive_key_from_passwd()` or something better.

Reply:
Sure, I think `derive_key_from_pw` is good.